inc/profile.class.php

```diff
@@ -494,6 +496,68 @@ static function getUnderActiveProfileRestrictRequest($separator = "AND") {
       return $query;
    }
 
+   /**
+    * Get SQL restrict criteria to determine profiles with less rights than the active one
+    *
+    * @since 9.3.1
+    *
+    * @return array
+   **/
+   static function getUnderActiveProfileRestrictCriteria() {
+
+      // Not logged -> no profile to see
+      if (!isset($_SESSION['glpiactiveprofile'])) {
+         return [0];
+      }
+
+      // Profile right : may modify profile so can attach all profile
+      if (Profile::canCreate()) {
+         return [1];
+      }
+
+      $criteria = ['glpi_profiles.interface' => Session::getCurrentInterface()];
+
+      // First, get all possible rights
+      $right_subqueries = [];
+      foreach (ProfileRight::getAllPossibleRights() as $key => $default) {
+         $val = isset($_SESSION['glpiactiveprofile'][$key])?$_SESSION['glpiactiveprofile'][$key]:0;
+
+         if (!is_array($val) // Do not include entities field added by login
+             && (Session::getCurrentInterface() == 'central'
+                 || in_array($key, self::$helpdesk_rights))) {
+            $right_subqueries[] = [
+               'glpi_profilerights.name'     => $key,
+               'glpi_profilerights.rights'   => ['|=', $val]
+            ];
+         }
+      }
+
+      $dbiterator = new DBmysqlIterator(null);
+      $dbiterator->execute(
```

Comment:
I would use `buildQuery` function here, to prevent anyone thinking that the query is executed.